rkt/status.go

```diff
@@ -16,22 +16,161 @@
 
 package main
 
-import "fmt"
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/appc/spec/schema/types"
+	"github.com/coreos/rocket/pkg/lock"
+)
 
 var (
 	cmdStatus = &Command{
 		Name:    "status",
 		Summary: "Check the status of a rkt job",
-		Usage:   "UUID",
+		Usage:   statusUsage,
 		Run:     runStatus,
 	}
+	flagWait bool
+)
+
+const (
+	statusDir   = "stage1/rkt/status"
+	statusUsage = "[--wait] UUID"
 )
 
 func init() {
 	commands = append(commands, cmdStatus)
+	cmdStatus.Flags.BoolVar(&flagWait, "wait", false, "toggle waiting for the container to exit if running")
 }
 
 func runStatus(args []string) (exit int) {
-	fmt.Println("Not implemented.")
+	if len(args) != 1 {
+		fmt.Fprintf(os.Stderr, "Usage: %s\n", statusUsage)
+		return 1
+	}
+
+	containerUUID, err := types.NewUUID(args[0])
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid UUID: %v\n", err)
+		return 1
+	}
+
+	cid := containerUUID.String()
+	isGarbage := false
+
+	cp := filepath.Join(containersDir(), cid)
+	l, err := lock.NewLock(cp)
+	if err != nil {
```

Comment:
I still don't think this is right because it catches any errors; e.g. in the case of `EACCES` we should fail immediately here, right? That's why I introduced `isNoSuchDirErr` in https://github.com/coreos/rocket/pull/304/files